File: daemon/src/routes/activity.ts
import { query } from './request'
import type { RouteTable } from './route'

export const activity = {
  /** What is at work in each checkout right now. Changes ride the socket; this is where a
   *  client that has just arrived reads the picture as it stands. */
  'GET /api/activity': (c, ctx) => c.json({ activity: ctx.activity }),

  /** Finished with a shell. Sockets do not end one — every way a socket has of closing is
   *  somebody meaning to come back — so this is where a tab says it is done. */
  'DELETE /api/terminal': (c, ctx) =>
    c.json({ closed: ctx.terminals.finish(query(c, 'session')) }),
} satisfies RouteTable

File: daemon/internal/api/activity.go
// What is at work in each checkout, and everything on this machine going away.

package api

import (
	"net/http"

	"github.com/broodmotherai/broodmother/daemon/internal/activity"
	"github.com/broodmotherai/broodmother/daemon/internal/app"
)

var activityTable = Table{
	/* Changes ride the socket; this is where a client that has just arrived reads the picture as
	   it stands. */
	"GET /api/activity": func(_ http.ResponseWriter, _ *http.Request, ctx *app.Context) (any, error) {
		states := map[string]activity.State{}
		if ctx.Activity != nil {
			states = ctx.Activity.States()
		}
		return map[string]any{"activity": states}, nil
	},

	/* Every folder in the home, gone. The last resort, and the one route that cannot be undone. */
	"DELETE /api/data": func(_ http.ResponseWriter, _ *http.Request, ctx *app.Context) (any, error) {
		saved, err := ctx.RemoveEverything()
		if err != nil {
			return nil, err
		}
		return map[string]any{"config": saved}, nil
	},
}
